pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
```

Comment:
What's your plan for the fix? We do have the `fspath` stuff now. It's possible we could just ensure that the path has been stringified before doing the compression inference.